GitUI/Editor/Diff/DiffViewerLineNumberCtrl.cs

```diff
@@ -112,19 +114,37 @@ public override void Paint(Graphics g, Rectangle rect)
             }
         }
 
-        private Dictionary<int, DiffLineNum> DiffLines { get; set; }
+        private ConcurrentDictionary<int, DiffLineNum> DiffLines { get; set; }
 
-        public void AddDiffLineNum(DiffLineNum diffLineNum)
+        public void AddDiffLineNum(List<DiffLineNum> diffLineNums)
         {
-            DiffLines[diffLineNum.LineNumInDiff] = diffLineNum;
-            _maxValueOfLineNum = Math.Max(diffLineNum.LeftLineNum, diffLineNum.RightLineNum);
+            foreach (var diffLineNum in diffLineNums)
+            {
+                DiffLines[diffLineNum.LineNumInDiff] = diffLineNum;
+                _maxValueOfLineNum = Math.Max(_maxValueOfLineNum,
+                    Math.Max(diffLineNum.LeftLineNum, diffLineNum.RightLineNum));
+            }
+            try
+            {
+                textArea.BeginInvoke(new Action(() =>
```

Comment:
This is not safe and causes exceptions. In the future please use the [InvokeAsync](https://github.com/gitextensions/gitextensions/blob/800caa27380cd053ddf2a8396f1b0e42352b7e60/GitUI/GitUIExtensions.cs#L301)